src/NHibernate.Test/NHSpecificTest/NH3403/Fixture.cs

```diff
@@ -0,0 +1,209 @@
+				Assert.AreEqual(expectedDbType, Driver.LastCommandParameters.First().SqlDbType);
+			}
+		}
+
+		[TestCase("Name", SqlDbType.NVarChar)]
+		//[TestCase("AnsiName", SqlDbType.VarChar)]// - Not yet possible
+		public void LinqStartsWithShouldUseLargerSize(string property, SqlDbType expectedDbType)
+		{
+			Driver.ClearCommands();
+
+			using (ISession session = OpenSession())
+			using (ITransaction transaction = session.BeginTransaction())
+			{
+				if (property == "Name")
+				{
+					session.Query<Entity>().Where(x => x.Name.StartsWith("Bob")).ToList();
+				}
+				else
+				{
+					session.Query<Entity>().Where(x => x.AnsiName.StartsWith("Bob")).ToList();
+				}
+				Assert.GreaterOrEqual(Driver.LastCommandParameters.First().Size, 4);
+				Assert.AreEqual(expectedDbType, Driver.LastCommandParameters.First().SqlDbType);
+
+			}
+		}
+
+		[TestCase("Name", SqlDbType.NVarChar)]
+		//[TestCase("AnsiName", SqlDbType.VarChar)]// - Not yet possible
+		public void LinqContainsShouldUseLargerSize(string property, SqlDbType expectedDbType)
+		{
+			Driver.ClearCommands();
+
+			using (ISession session = OpenSession())
+			using (ITransaction transaction = session.BeginTransaction())
+			{
+				if (property == "Name")
+				{
+					session.Query<Entity>().Where(x => x.Name.Contains("Bob")).ToList();
+				}
+				else
+				{
+					session.Query<Entity>().Where(x => x.AnsiName.Contains("Bob")).ToList();
+				}
+				Assert.GreaterOrEqual(Driver.LastCommandParameters.First().Size, 5);
+				Assert.AreEqual(expectedDbType, Driver.LastCommandParameters.First().SqlDbType);
+
+			}
+		}
+
+		[TestCase("Name", SqlDbType.NVarChar)]
+		[TestCase("AnsiName", SqlDbType.VarChar)]
+		public void HqlLikeShouldUseLargerSize(string property, SqlDbType expectedDbType)
+		{
+			Driver.ClearCommands();
+
+			using (ISession session = OpenSession())
+			using (ITransaction transaction = session.BeginTransaction())
+			{
+				session.CreateQuery("from Entity where " + property + " like :name").SetParameter("name","%Bob%").List<Entity>();
+
+				Assert.GreaterOrEqual(Driver.LastCommandParameters.First().Size, 5);
+				Assert.AreEqual(expectedDbType, Driver.LastCommandParameters.First().SqlDbType);
+			}
+		}
+
+		[TestCase("Name", SqlDbType.NVarChar)]
+		[TestCase("AnsiName", SqlDbType.VarChar)]
+		public void CriteriaEqualsShouldUseMappedSize(string property, SqlDbType expectedDbType)
+		{
+			Driver.ClearCommands();
+
+			using (ISession session = OpenSession())
+			using (ITransaction transaction = session.BeginTransaction())
+			{
+				Driver.ClearCommands();
+
+				session.CreateCriteria<Entity>().Add(Restrictions.Eq(property, "Bob"))
+								  .List<Entity>();
+
+				Assert.GreaterOrEqual(Driver.LastCommandParameters.First().Size, 3);
+				Assert.AreEqual(expectedDbType, Driver.LastCommandParameters.First().SqlDbType);
+			}
+		}
+
+		[TestCase("Name")]
+		[TestCase("AnsiName")]
+		public void CriteriaEqualsWithTooLargeValueShouldThrow(string property)
```

Comment:
Um... I believe the current released behaviour is to NOT throw an exception (in practice), because the lengths used by SqlClientDriver would usually exceed the length of the entered values, and the other drivers don't set a length, so they would also just use the entered value. Am I mistaken?